### DISCUSSION TITLE
Edit gcode in IOSender

### DISCUSSION THREAD

**Discussion by raenji-sk** [terjeio/ioSender#100] [Ideas]
Would it be possible to add an option for editing the opened gcode file directly in the IOSender, or create a new gcode there? 
Again comparing it to bCNC where you can edit the gcode and also quickly create some simple code.
Not saying its a super necessary function, but it is quite nice for quick tests and correcting possible mistakes without having to open the file in an external editor or running the whole thing through the postprocessor again.

**Comment by terjeio** [comment]
I have added this to the [roadmap](https://github.com/terjeio/ioSender/wiki/Roadmap).

For now you may use the _Edit file_ tool in the toolbar, this opens the file in Notepad for editing. When saving the file and exiting Notepad the file will be automatically reloaded. It is possible to use another editor by editing the _App.config file_, I need to add a UI setting for that.

**Reply by raenji-sk** [reply]
OK, that works for now. 
If you add the direct editing feature in the future, would the 3D view be updated in realtime to show the new or modified gcode from the "Program" tab?

**Reply by terjeio** [reply]
> If you add the direct editing feature in the future, would the 3D view be updated in realtime to show the new or modified gcode from the "Program" tab?

For relatively small files this would be possible, for larger ones that takes seconds to display IMO not so.

**Reply by raenji-sk** [reply]
Seconds is OK, I mean it does not need to be realtime. Just to be able to see the changes in the 3D viewer, if it takes a couple of seconds to refresh then its OK.
